### ISSUE TITLE
OptionsRESTController is registered with incorrect schema argument

### ISSUE THREAD

**Issue by brandonpayton** [Automattic/amp-wp#5281]
## Bug Description

OptionsRESTController registers a route with an incorrect schema argument. It [should pass a schema callback](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#resource-schema) but [passes data instead](https://github.com/ampproject/amp-wp/blob/f44c07f606713ee78e6e4ce0d437749c5821c8f8/src/OptionsRESTController.php#L138).

This appears to trigger a warning in WordPress 5.5:
```
Warning: call_user_func() expects parameter 1 to be a valid callback, array must have exactly two members in /wordpress/core/5.5/wp-includes/rest-api/class-wp-rest-server.php on line 1279
```

I am seeing this when using WP-CLI v2.4.0, specifically when running `wp shell`. When I adjust the OptionsRESTController schema argument to be a callback, the warning goes away.

## Expected Behaviour

OptionsRESTController registers its route with a schema callback, and the related warning goes away.

## Steps to reproduce

With a test site:
1. Install v2.0.0 of the AMP plugin
2. Use WP-CLI v2.4.0
3. Run `wp shell`

## Additional context

 - WordPress version: 5.5
 - Plugin version: 2.0.0
 - Gutenberg plugin version (if applicable): n/a
 - AMP plugin template mode: Reader
 - PHP version: 7.3.21
 - OS: Linux
 - Browser: n/a
 - Device: n/a

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
@brandonpayton do you have a patch with what you did that you can open in a PR?

**Comment by westonruter** [comment]
I wasn't able to reproduce this issue via `wp shell`, but I did get a PHP warning when doing `curl --request OPTIONS https://wordpressdev.lndo.site/wp-json/amp/v1/options`:

```
PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, array must have exactly two members in /app/public/core-dev/src/wp-includes/rest-api/class-wp-rest-server.php on line 1279
```

Submitted fix in #5283.

**Comment by brandonpayton** [comment]
> @brandonpayton do you have a patch with what you did that you can open in a PR?

Hi @westonruter, I'd actually opened the PR creation page for [this change](https://github.com/brandonpayton/amp-wp/commit/617a803a23ff14e68b0e3eb771c69425fb09669a), but I ended up leaving it to you since I wasn't clear about signing the CLA and figured it would be a quick fix on your end.

Thank you for working through the issue and fixing.
